packages/core/data-transfer/src/strapi/providers/local-source/assets.ts

```diff
@@ -1,36 +1,40 @@
 import { join } from 'path';
-import https from 'https';
-import http from 'http';
 import { Duplex, PassThrough, Readable } from 'stream';
 import { stat, createReadStream, ReadStream } from 'fs-extra';
 import type { LoadedStrapi } from '@strapi/types';
 
 import type { IAsset } from '../../../../types';
 
-const protocolForPath = (filepath: string) => {
-  return filepath?.startsWith('https') ? https : http;
-};
-
 function getFileStream(filepath: string, isLocal = false): PassThrough | ReadStream {
   if (isLocal) {
     return createReadStream(filepath);
   }
 
   const readableStream = new PassThrough();
 
-  protocolForPath(filepath)
-    .get(filepath, (res) => {
-      if (res.statusCode !== 200) {
-        readableStream.emit(
-          'error',
-          new Error(`Request failed with status code ${res.statusCode}`)
-        );
+  fetch(filepath)
+    .then((res) => {
+      if (res.status !== 200) {
+        readableStream.emit('error', new Error(`Request failed with status code ${res.status}`));
         return;
       }
 
-      res.pipe(readableStream);
+      // Use pipeTo to transfer it from the Response.body to the readable stream
+      const writable = new WritableStream({
+        write(chunk) {
+          readableStream.write(chunk);
+        },
+        close() {
+          readableStream.end();
+        },
+        abort() {
+          readableStream.destroy();
+        },
+      });
+
+      res?.body?.pipeTo(writable);
```

Comment:
Can't you just pipeTo the passthrough? I believe it extends both writable & readable streams

Reply:
![image](https://github.com/strapi/strapi/assets/25851739/1cf93ddf-48c5-4feb-bfdc-66b9070718a4)

Reply:
Summary of offline discoveries regarding :arrow_up: 
- fetch is returning a libdom.ReadableStream (`source`) which is not compatible with stream.Writable(`dest`)
- Using stream.Readable.fromWeb(`source`).pipe(`dest`) doesn't work as fromWeb expects a stream/web.ReadableStream
- Using stream.Readable.fromWeb(new stream/web.ReadableStream(`source`)).pipe(`dest`) seems to be the best option

Reply:
This should be resolved now, thanks!